parcel.go

```diff
@@ -2,59 +2,106 @@ package main
 
 import (
 	"database/sql"
+	"fmt"
+
+	_ "modernc.org/sqlite"
 )
 
 type ParcelStore struct {
 	db *sql.DB
 }
 
-func NewParcelStore(db *sql.DB) ParcelStore {
+func NewParcelStore(db *sql.DB) ParcelStore { // возвращает посылки
 	return ParcelStore{db: db}
 }
 
+// добавляет объект Parcel в хранилище ParcelStore и возвращает идентификатор добавленного объекта и ошибку, если она произошла
 func (s ParcelStore) Add(p Parcel) (int, error) {
 	// реализуйте добавление строки в таблицу parcel, используйте данные из переменной p
 
+	res, err := s.db.Exec("INSERT INTO parcel (client, status, address, created_at) VALUES (?, ?, ?, ?)",
+		p.Client, p.Status, p.Address, p.CreatedAt)
+	if err != nil {
+		return 0, err
+	}
+
 	// верните идентификатор последней добавленной записи
-	return 0, nil
+	id, err := res.LastInsertId()
+	if err != nil {
+		return 0, err
+	}
+	return int(id), nil
+
 }
 
 func (s ParcelStore) Get(number int) (Parcel, error) {
 	// реализуйте чтение строки по заданному number
 	// здесь из таблицы должна вернуться только одна строка
 
 	// заполните объект Parcel данными из таблицы
+	row := s.db.QueryRow("SELECT number, client, status, address, created_at FROM parcel WHERE number = ?", number)
+
 	p := Parcel{}
+	// Сканируем результат запроса
+	err := row.Scan(&p.Number, &p.Client, &p.Status, &p.Address, &p.CreatedAt)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			return Parcel{}, fmt.Errorf("посылка с номером %d не найдена", number)
+		}
+		return Parcel{}, fmt.Errorf("ошибка при получении посылки: %w", err)
+	}
+
+	// Заполняем объект Parcel данными из таблицы
+	p.Number = number // Устанавливаем номер посылки
 
 	return p, nil
 }
 
 func (s ParcelStore) GetByClient(client int) ([]Parcel, error) {
 	// реализуйте чтение строк из таблицы parcel по заданному client
 	// здесь из таблицы может вернуться несколько строк
+	rows, err := s.db.Query("SELECT number, client, status, address, created_at FROM parcel WHERE client = ?", client)
+	if err != nil {
+		return nil, fmt.Errorf("ошибка при выполнении запроса: %v", err)
+	}
+	defer rows.Close() // Закрываем rows после завершения работы с ними
 
 	// заполните срез Parcel данными из таблицы
 	var res []Parcel
-
+	// Проходим по всем строкам результата запроса
+	for rows.Next() {
+		var p Parcel
+		// Сканируем данные в объект Parcel
+		err := rows.Scan(&p.Number, &p.Client, &p.Status, &p.Address, &p.CreatedAt)
+		if err != nil {
+			return nil, fmt.Errorf("ошибка при сканировании строки: %v", err)
```

Comment:
При оборачивании нужно использовать `%w` (wrap), а просто при выводе в консоль `%v`
```suggestion
			return nil, fmt.Errorf("ошибка при сканировании строки: %w", err)
```

Reply:
Нужно везде поправить